sdk/src/record-provider.ts

```diff
@@ -8,17 +8,72 @@ import { AleoNetworkClient } from "./network-client.js";
  * implementations.
  */
 interface RecordSearchParams {
+    unspent: boolean;
     [key: string]: any; // This allows for arbitrary keys with any type values
 }
 
+type RecordsResponseFilter = {
+    program: boolean;
+    record: boolean;
+    function: boolean;
+    transition: boolean;
+    blockHeight: boolean;
+    transactionId: boolean;
+    transitionId: boolean;
+    ioIndex: boolean;
+}
+
+type EncryptedRecord = {
+    commitment: string;
+    checksum?: string;
+    blockHeight?: number;
+    programName?: string;
+    functionName?: string;
+    outputIndex?: number;
+    owner?: string;
+    recordCiphertext?: string;
+    recordName?: string;
+    recordNonce?: string;
+    transactionId?: string;
+    transitionId?: string;
+    transactionIndex?: number;
+    transitionIndex?: number;
+}
+
+type OwnedRecord = {
+    blockHeight?: number;
+    commitment?: string;
+    functionName?: string;
+    outputIndex?: number;
+    owner?: string;
+    programName?: string;
+    recordCiphertext?: string;
+    recordPlaintext?: string;
+    recordName?: string;
+    spent?: boolean;
+    tag?: string;
+    transactionId?: string;
+    transitionId?: string;
+    transactionIndex?: number;
+    transitionIndex?: number;
+}
+
+interface RecordResponse {
+    owner?: string;
+}
+
 /**
  * Interface for a record provider. A record provider is used to find records for use in deployment and execution
  * transactions on the Aleo Network. A default implementation is provided by the NetworkRecordProvider class. However,
  * a custom implementation can be provided (say if records are synced locally to a database from the network) by
  * implementing this interface.
  */
 interface RecordProvider {
-    account: Account
+    getEncryptedRecords(recordsFilter: RecordSearchParams, responseFilter: RecordsResponseFilter): Promise<EncryptedRecord[]>;
+
+    serialNumbersExist(serialNumbers: string[]): Promise<Record<string, boolean>>;
```

Comment:
I think the return type on this is wrong, Record in the SDK is a wasm type. The return type should be an object with a list of commitment strings with a boolean indicating it's spent or not.

You could create a type for it in the `models` directory like this:

export type SpentResponse = { [tag: string]: boolean };

Also, can you put a doc comment on this and all methods?

```suggestion
    checkSerialNumbers(serialNumbers: string[]): Promise<SpentResponse>;
```

Reply:
This is different, Record is a ts type, its basically a map